tools/config-gen/params.yaml

```diff
@@ -15,16 +31,308 @@
+    by Ctrl+C). This prevents those signals from immediately terminating gcsfuse
+    inflight operations.
+
+- flag-name: "custom-endpoint"
+  config-path: "gcs-connection.custom-endpoint"
+  type: "url"
+  usage: >-
+    Specifies an alternative custom endpoint for fetching data. Should only be
+    used for testing.  The custom endpoint must support the equivalent resources
+    and operations as the GCS  JSON endpoint,
+    https://storage.googleapis.com/storage/v1. If a custom endpoint is not
+    specified,  GCSFuse uses the global GCS JSON API endpoint,
+    https://storage.googleapis.com/storage/v1.
+
+- flag-name: "anonymous-access"
+  config-path: "gcs-auth.anonymous-access"
+  type: "bool"
+  usage: "Authentication is enabled by default. This flag disables authentication"
+
+- flag-name: "billing-project"
+  config-path: "gcs-connection.billing-project"
+  type: "string"
+  usage: >-
+    Project to use for billing when accessing a bucket enabled with "Requester
+    Pays". (The default is none)
+
+- flag-name: "key-file"
+  config-path: "gcs-auth.key-file"
+  type: "resolvedPath"
+  usage: "Absolute path to JSON key file for use with GCS. (The default is none, Google application default credentials used)"
+
+- flag-name: "token-url"
+  config-path: "gcs-auth.token-url"
+  type: "string"
+  usage: "A url for getting an access token when the key-file is absent."
+
+- flag-name: "reuse-token-from-url"
+  config-path: "gcs-auth.reuse-token-from-url"
+  type: "bool"
+  usage: "If false, the token acquired from token-url is not reused."
   default: "true"
+
+- flag-name: "limit-bytes-per-sec"
+  config-path: "gcs-connection.limit-bytes-per-sec"
+  type: "float64"
+  usage: "Bandwidth limit for reading data, measured over a 30-second window. (use -1 for no limit)"
+  default: "-1"
+
+- flag-name: "limit-ops-per-sec"
+  config-path: "gcs-connection.limit-ops-per-sec"
+  type: "float64"
+  usage: "Operations per second limit, measured over a 30-second window (use -1 for no limit)"
+  default: "-1"
+
+- flag-name: "sequential-read-size-mb"
+  config-path: "gcs-connection.sequential-read-size-mb"
+  type: "int"
+  usage: "File chunk size to read from GCS in one call. Need to specify the value in MB. ChunkSize less than 1MB is not supported"
+  default: "200"
+
+- flag-name: "max-retry-sleep"
+  config-path: "gcs-retries.max-retry-sleep"
+  type: "duration"
+  usage: >-
+    The maximum duration allowed to sleep in a retry loop with exponential
+    backoff for failed requests to GCS backend. Once the backoff duration
+    exceeds this limit, the retry continues with this specified maximum value.
+  default: "30s"
+
+- flag-name: "stat-cache-max-size-mb"
+  config-path: "metadata-cache.stat-cache-max-size-mb"
+  type: "int"
+  usage: >-
+    The maximum size of stat-cache in MiBs. It can also be set to -1 for
+    no-size-limit, 0 for no cache. Values below -1 are not supported.
+  default: "math.MinInt64"
```

Comment:
shouldn't this be 32?

Reply:
Sure, I've set it to 32. In the current code, it's currently set to math.MinInt64 initially by default and later on it's value is resolved based on this and some other flags. We will have to change that logic before we start using this new config. Same for metadata-cache-ttl.